### PR TITLE
Fix failing GameDetailsModal tests from PR #893 (Files tab)

### PR DESCRIPTION
**Pull request by Doezer**
## Description

PR #893 ("Add Game Details files tab") had a failing `test (3)` CI shard: the new `File`/`Folder` lucide-react imports added for the Files tab weren't present in the test file's `lucide-react` mock, so every test in `GameDetailsModal.test.tsx` failed at render (33 of 34 tests). This PR merges #893's changes into `main` and fixes the test breakage, plus addresses an outstanding review-tooling finding on the same file.

This branch is intended to replace/supersede the failing state of #893.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Changes

- Added `File` and `Folder` to the `lucide-react` mock in `client/__tests__/GameDetailsModal.test.tsx` so the Files tab renders correctly under test.
- Added new test coverage for the Files tab: loading state, error state, empty state, flat file list, grouped-by-category list, and that the tab is hidden for discovery (unsaved) games.
- Replaced the array-index `key` with `file.path` (a stable, unique value) in the two file-list renders — the last unresolved CodeRabbit finding on #893 (the accompanying Tailwind-spacing part of that finding had already been addressed).

## Validation

- `npm run check` — passes
- `npx prettier --check` — passes
- `npx eslint` on changed files — passes
- `npm run test:run` — full suite passes: 2191 passed, 7 skipped

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] If this PR adds a new actor/integration, external interface, or security-relevant change, I have updated docs/ARCHITECTURE.md, docs/API.md, and/or docs/SECURITY_ASSESSMENT.md accordingly
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

---
🤖 Generated with [Claude Code](https://claude.ai/code)

---
_Generated by [Claude Code](https://claude.ai/code/session_016kQoX2zyFhaxxvicYPhbyn)_

## Summary by CodeRabbit

* **New Features**
  * Added a **Files** tab to game details for supported games.
  * Displays scanned files in flat lists or grouped by category, with file icons.
  * Includes loading, error, and empty-state messages.
  * Automatically returns to the Overview tab when switching games.
  * Hides the Files tab for discovery games.
  * Correctly categorizes files, including additional content and packs.

### CODE REVIEW

**Comment by Doezer** [comment]
>  **wshm** · Automated triage by AI

## 📊 Automated PR Analysis

| | |
|---|---|
| 🐛 **Type** | `bug-fix` |
| 🟢 **Risk** | `low` |

### Summary

This PR merges the previously failing PR #893 (Add Game Details files tab) into main and fixes the CI test failure caused by missing lucide-react mocks for File/Folder icons. It also adds new test coverage for the Files tab (loading, error, empty, flat/grouped list, discovery-game hidden state) and replaces array-index keys with stable file.path keys as requested in an outstanding code review finding.

### Review Checklist
- [x] Tests present
- [ ] Breaking change
- [ ] Docs updated

---
*Analyzed automatically by [wshm](https://github.com/wshm-dev/wshm)* · This is an automated analysis, not a human review.

File: client/src/components/GameDetailsModal.tsx [hunk 1]
```diff
@@ -609,6 +616,37 @@ export default function GameDetailsModal({ game, open, onOpenChange }: GameDetai
...
+    isLoading: filesLoading,
+    isError: filesError,
+  } = useQuery<Array<{ name: string; path: string; category: string; isDirectory: boolean }>>({
+    queryKey: [`/api/games/${game?.id}/files`],
+    queryFn: async () => {
+      const res = await apiRequest("GET", `/api/games/${game!.id}/files`);
+      const data = await res.json();
+      return data.files;
+    },
+    enabled: open && !!game?.id && !isDiscoveryId(game.id),
```
[the remark this answers is not in the document]
Comment (Doezer) [reply client/src/components/GameDetailsModal.tsx]: Fixed in 5ae27cc — the query's `enabled` now requires `activeTab === "files"`, so the recursive filesystem scan only fires once the user actually opens the Files tab.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: client/src/components/GameDetailsModal.tsx [hunk 2]
```diff
@@ -609,6 +616,37 @@ export default function GameDetailsModal({ game, open, onOpenChange }: GameDetai
...
+    data: gameFiles = [],
+    isLoading: filesLoading,
+    isError: filesError,
+  } = useQuery<Array<{ name: string; path: string; category: string; isDirectory: boolean }>>({
+    queryKey: [`/api/games/${game?.id}/files`],
+    queryFn: async () => {
+      const res = await apiRequest("GET", `/api/games/${game!.id}/files`);
+      const data = await res.json();
+      return data.files;
+    },
```
[the remark this answers is not in the document]
Comment (Doezer) [reply client/src/components/GameDetailsModal.tsx]: Fixed in 5ae27cc + 0d82217: removed the `isDirectory` field and the dead `Folder`-icon branch (the endpoint never returns directory entries), and defined the response contract as `ScannedGameFile` in `shared/schema.ts`, now used by both the endpoint and the client query instead of two independently hand-written shapes.

---
_Generated by [Claude Code](https://claude.ai/code)_
